src/js/first-run/first-run.js

```diff
@@ -12,10 +40,55 @@ document.addEventListener("DOMContentLoaded", async () => {
 
   const oauthEntryPoints = document.querySelectorAll(".open-oauth");
   oauthEntryPoints.forEach(el => {
-    el.addEventListener("click", (e) => {
+    el.addEventListener("click", async(e) => {
       e.preventDefault();
       sendRelayEvent("First Run", "click", e.target.dataset.eventLabel);
-      return window.open(`${relaySiteOrigin}/accounts/profile?utm_source=fx-relay-addon&utm_medium=first-run&utm_content=first-run-sign-up-btn`);
+      const authUrl = `${fxaAuthBaseUrl}/authorization`;
+      const scope = 'profile';
+      const accessType = 'offline';
+      const redirectUrl = browser.identity.getRedirectURL();
+      const state = Math.random().toString(36).substr(2);
+      const pkceCodeVerifier = generatePKCECodeVerifierValue();
+      console.log(`pkceCodeVerifier: ${pkceCodeVerifier}`);
+      const pkceCodeChallenge = await pkceCodeChallengeB64(pkceCodeVerifier);
+      console.log(`pkceCodeCallenge: ${pkceCodeChallenge}`);
+      const fullAuthUrl = `${authUrl}?client_id=${clientID}&scope=${scope}&access_type=${accessType}&redirect_uri=${redirectUrl}&state=${state}&code_challenge_method=S256&code_challenge=${pkceCodeChallenge}&response_type=code`;
```

Comment:
Suggestion: Wrap this entire set of declarations into a helper function (not inside this click function)? (Eg: `buildFullAuthUrl()`, etc.)